modules/nf-core/sentieon/tnhaplotyper2/tests/main.nf.test

```diff
@@ -0,0 +1,293 @@
+                    process.out.contamination_segments,
+                    process.out.versions,
+                    path(process.out.versions[0]).yaml
+                    ).match() }
+            )
+        }
+    }
+
+    test("tumour only") {
+
+        when {
+            params {
+                module_args  = ''
+                module_args2 = '--tumor_sample tumour'
+                module_args3 = '--tumor_sample tumour'
+                module_args4 = '--tumor_sample tumour'
+            }
+            process {
+                """
+                input[0] = TUMOUR_ADDORREPLACEREADGROUPS.out.bam
+                    .join(TUMOUR_ADDORREPLACEREADGROUPS.out.bai)
+                    .map{
+                    meta,tumour_bam,tumour_bai->
+                    [meta,tumour_bam,tumour_bai,[]]
+                    }
+                input[1] = [[:],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.dict', checkIfExists:true)]
+                input[2] = [[:],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta', checkIfExists:true)]
+                input[3] = [[:],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai', checkIfExists:true)]
+                input[4] = [[:],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/gnomAD.r2.1.1.vcf.gz', checkIfExists:true)]
+                input[5] = [[:],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/gnomAD.r2.1.1.vcf.gz.tbi', checkIfExists:true)]
+                input[6] = [[:],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/mills_and_1000G.indels.hg38.vcf.gz', checkIfExists:true)]
+                input[7] = [[:],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/mills_and_1000G.indels.hg38.vcf.gz.tbi', checkIfExists:true)]
+                input[8] = true
+                input[9] = true
+                """
+            }
+        }
+        then {
+            assert process.success
+            assertAll(
+                { assert snapshot(
+                    path(process.out.vcf[0][1]).vcf.summary,
+                    file(process.out.index[0][1]).name,
+                    process.out.versions,
+                    process.out.stats,
+                    path(process.out.versions[0]).yaml
+                    ).match() }
+            )
+        }
+    }
+
+
+    test("tumour normal - stub") {
+        options '-stub'
+        when {
+            params {
+                module_args  = ''
+                module_args2 = '--tumor_sample tumour --normal_sample normal'
+                module_args3 = ''
+                module_args4 = ''
+            }
+            process {
+                """
+                input[0] = [[id:'test'],[],[],[]]
```

Comment:
Having empty channel for the stub test would be beneficial for the whole nf-core repository as it would speed up all the test and reduce the computational load.
We already discussed that with Maxime, and I think it would be worth discussing about its implementation.